### DISCUSSION TITLE
How can we restrict file upload to certain file types?

### DISCUSSION THREAD

**Discussion by hellosmriti** [pimcore/pimcore#7835] [Q&A / Support] [answered]
How can we restrict file upload to certain file types only like for asset we wish to restrict upload extension to .jpg/.jpeg only or upload MIME type to image/jpeg?

**Comment by NiklasBr** [comment, marked as answer]
I suppose you might be able to use the `AssetEvents::PRE_ADD` [event](https://pimcore.com/docs/pimcore/current/Development_Documentation/Extending_Pimcore/Event_API_and_Event_Manager.html) and inspect the file type then and there.
